test/integration/minishift.go

```diff
@@ -190,6 +190,43 @@ func (m *Minishift) executingMinishiftCommand(command string) error {
 	return nil
 }
 
+func (m *Minishift) setImageCaching(operation string) error {
+	enabled := "true"
+	if operation == "disabled" {
+		enabled = "false"
+	}
+
+	return m.executingMinishiftCommand(fmt.Sprintf("config set image-caching %s", enabled))
+}
+
+func (m *Minishift) imageExportShouldComplete(noOfImages int) error {
+	// poll till the output of the `minishift image list` shows three OpenShift cached images
+	timeout := time.NewTimer(20 * time.Minute)
+
+outerPollActive:
+	for {
+		select {
+		case <-timeout.C:
+			return errors.New("Timed out in getting the number of default cached images")
+		default:
+			cmdOut, _, _ := m.runner.RunCommand("image list")
+			cmdOut = strings.TrimRight(cmdOut, "\n")
+			if len(strings.Split(cmdOut, "\n")) == noOfImages {
```

Comment:
If higher than expected I think we could return error?

Reply:
It will break from the loop whenever it reaches `noOfImages` but it would be worthing thinking if `image list` will give me result greater than `noOfImages` then whether we should break out of loop or return error.

Reply:
@budhrg If the number of images is greater than noOfImages I would suggest to error as we clearly define we expect it to be for example 3 images. However it is quite possible that even if it would download 4 images, they will not finish in same window of 5 seconds and it will still pass.